File: backend/app/model/us_examination_neural_model.py
from pydantic import BaseModel


class USExaminationNeuralModel(BaseModel):
    id: str
    title: str
    port: int
    description: dict[str, str]

    def get_localized_description(self, language_code: str) -> str:
        return self.description.get(language_code) or next(iter(self.description.values()), "")

File: ios/Doglyad/Domain/Ultrasound/USExaminationNeuralModel.swift
import Foundation

struct USExaminationNeuralModel: Codable, Identifiable, Equatable {
    let id: String
    let title: String
    let description: [String: String]

    func getLocalizedDescription(for locale: Locale) -> LocalizedStringResource {
        let key = locale.language.languageCode?.identifier ?? "en"
        let description = description[key] ?? description.values.first ?? ""
        return LocalizedStringResource(stringLiteral: description)
    }
}
